### PR TITLE
Landmine Rework

### LINKED ISSUE #1861: Directional Lighting Sets Off Landmines
Issue reported from  Round ID: 1468 (Shiptest)
image

Reporting client version: 514.1589

### PR DESCRIPTION
**Pull request by Sun-Soaked**
## About The Pull Request
Comprehensive rework of landmines.

- [x] Turn mines into an item
- [x] Port a bunch of tg improvements
- [x] Port tg shrapnel fixes
- [x] Add inhand mine deploying
- [x] Mines now take time to arm
- [x] Add mine disarming
- [x] Delayed explosion
- [x] fix shrapnel runtime 
- [x] Actually MOVE the Mines file to code/items
- [x] Write a bunch of new mine types
- [x] Implement new mines in spawners
- [x] YOU THOUGHT IT WAS OVER. Proximity Mines & full pressure mine code edit
- [x] Embeds now work as intended

- [ ] Sprites (need help)

## Why It's Good For The Game
Pipe bomb....
Crew mortality grew two sizes that day

## Changelog

:cl:
add: Redesigns landmines as an item that can be picked up, disarmed, deployed. Some logic courtesy of /tg/.
add: Adds a proximity explosive variant of landmines.
fix: shrapnel embedding once again respects targeted limb and no longer runtimes on hitting a wall or destroying an object.
fix: storage can no longer pick up anchored objects, causing them to get stuck inside
balance: Nerfs the ability of heavy explosions to immediately breach to space
soundadd: adds two mine detonation noises
code: moves weapon manufacturer defines to a DEFINE file
/:cl:

### CODE REVIEW

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -1,162 +1,433 @@
...
+				else
+					user.visible_message(span_danger("[user] attempts to pick up \the [src] only to hear a beep as it explodes in \his hands!"), span_userdanger("You attempt to pick up \the [src] only to hear a beep as it explodes in your hands!"))
+					triggermine()
+					update_appearance(UPDATE_ICON_STATE)
+					return
+	. =..()
+
+//handles disarming(and failing to disarm)
+/obj/item/mine/attackby(obj/item/I, mob/user)
+	if(armed)
```
Comment (west3436) [inline code/game/objects/effects/mines.dm]: Save an indent level via:
`if(!armed)`
`     to_chat(user, span_notice("You hit \the [src] with [I]. Thankfully, nothing happens."))`
`return`

Comment (Sun-Soaked) [reply code/game/objects/effects/mines.dm]: true

**Comment by Sun-Soaked** [comment]
should be feature complete now. Just needs sprites and I'll undraft

File: code/game/objects/items/devices/mines.dm [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+//let them know the mine's done cooking
+/obj/item/mine/proc/now_armed()
+	armed = TRUE
+	alpha = stealthpwr
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/machines/nuke/angry_beep.ogg', 40, FALSE, -2)
+	visible_message("<span class='danger'>\The [src] beeps softly, indicating it is now active.<span>", vision_distance = COMBAT_MESSAGE_RANGE)
+
+//insert your horrible fate here
+/obj/item/mine/proc/mineEffect(mob/victim)
```
Comment (thgvr) [inline code/game/objects/items/devices/mines.dm]:
```suggestion
/obj/item/mine/proc/mine_effect(mob/victim)
```

File: code/game/objects/items/devices/mines.dm [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+	if(!can_trigger(gone))
+		return
+	// Check that the guy who's on it is stepping off
+	if(foot_on_mine && !IS_WEAKREF_OF(gone, foot_on_mine))
+		return
+	INVOKE_ASYNC(src, PROC_REF(triggermine), gone)
+	foot_on_mine = null
+
+/// When something sets off a mine
+/obj/item/mine/proc/triggermine(atom/movable/triggerer)
```
Comment (thgvr) [inline code/game/objects/items/devices/mines.dm]:
```suggestion
/obj/item/mine/proc/trigger_mine(atom/movable/triggerer)
```

File: code/game/objects/items/devices/mines.dm [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+		var/mob/living/defuser = foot_on_mine.resolve()
+		defuser.remove_movespeed_modifier(/datum/movespeed_modifier/stepped_on_mine)
+		foot_on_mine = null
+	update_appearance(UPDATE_ICON_STATE)
+	return
+
+//using an unarmed mine inhand deploys it.
+/obj/item/mine/attack_self(mob/user)
+	if(!armed)
+		user.visible_message("<span class='danger'>[user] deploys the [src].</span>", "<span class='notice'>You deploy the [src].</span>")
```
Comment (thgvr) [inline code/game/objects/items/devices/mines.dm]:
```suggestion
		user.visible_message(span_danger("[user] deploys the [src].</span>"), span_notice("You deploy the [src].</span>"))
```

**Comment by Sun-Soaked** [comment]
todo in future pr: more fitting sprites

**Pull request merged**